src/Purl/Fragment.php

```diff
@@ -109,7 +111,17 @@ public function __toString() : string
     protected function doInitialize() : void
     {
         if ($this->fragment !== null) {
-            $this->data = array_merge($this->data, parse_url($this->fragment));
+            $pos   = strpos($this->fragment, ':', 1);
```

Comment:
Here is my approach:
- Detect the hash bang whether it's with the colon.
- Rebuild the hash bang with the colon after using the ```parse_url``` function.

Reply:
@peter279k: Are we sure that this is the only case that would cause `parse_str` to fail ? Why not just use `parse_str` right away, and fallback to  `$data = ['path' => $this->fragment]` if it fails ?

It would make less assumptions about how `parse_str` works now and in the future. This would make this code more robust.